buffer/src/main/java/io/netty/buffer/ByteBuf.java

```diff
@@ -1736,6 +1736,18 @@ public double readDoubleLE() {
      */
     public abstract CharSequence readCharSequence(int length, Charset charset);
 
+    /**
+     * Gets a {@link String} with the given length at the current {@code readerIndex}
+     * and increases the {@code readerIndex} by the given length.
+     *
+     * @param length the length to read
+     * @param charset that should be used
+     * @return the string
+     * @throws IndexOutOfBoundsException
+     *         if {@code length} is greater than {@code this.readableBytes}
+     */
+    public abstract String readString(int length, Charset charset);
```

Comment:
This is a breaking change... you will need to add an implementation here. 

```suggestion
    public String readString(int length, Charset charset) {
        String string = toString(readerIndex(), length, charset);
        skipBytes(length);
        return string;
    }
```